src/Illuminate/Queue/SerializesAndRestoresModelIdentifiers.php

```diff
@@ -66,7 +69,9 @@ protected function getRestoredPropertyValue($value)
     protected function restoreCollection($value)
     {
         if (! $value->class || count($value->id) === 0) {
-            return new EloquentCollection;
+            return ! is_null($value->collectionClass ?? null)
```

Comment:
The `?? null` part here is a BC measure to make sure we don't break already serialized model identifiers instances that don't have this property.